gulpfile.js

```diff
@@ -1,23 +1,120 @@
 var gulp = require('gulp');
+var fs = require('fs');
 var plugins = require('gulp-load-plugins')();
 var del = require('del');
 var glob = require('glob');
 var path = require('path');
+var marked = require('marked');
+var swig = require('swig');
+var through2 = require('through2');
+var promisify = require('es6-promisify');
+var assert = require('assert');
+var mergeStream = require('merge-stream');
+var renderFile = promisify(swig.renderFile);
+var fsWriteFile = promisify(fs.writeFile);
+var fsReadFile = promisify(fs.readFile);
 
-var ignore = ['!./dist', '!./dist/**', '!./node_modules', '!./node_modules/**'];
-var recipes = glob.sync('./!(dist|node_modules)/').map(function toBase(dir) {
+var ignore = ['!./dist', '!./dist', '!./dist/**', '!./node_modules', '!./node_modules/**'];
+var recipeSlugs = glob.sync('./!(dist|node_modules|src)/').map(function toBase(dir) {
   return path.basename(dir);
 });
-var srcRecipes = recipes.map(function makePath(name) {
+var srcRecipes = recipeSlugs.map(function makePath(name) {
   return './' + name + '/**';
 });
-var srcFiles = recipes.map(function makeSourceFile(name) {
-  return './' + name + '/*.js';
-});
+
+var recipes = (function() {
+  var cache;
+  return function() {
+    if (cache) {
+      return cache;
+    }
+    cache = [];
+    recipeSlugs.forEach(function(recipe) {
+      var tokens = marked.lexer(fs.readFileSync(recipe + '/README.md', 'utf8'));
+      if (!tokens.length) {
+        console.warn(recipe + ': is missing title');
+      }
+      var name = tokens[0].text;
+      if (!name.startsWith('Recipe: ')) {
```

Comment:
I just needed it to be consistent so titles didn't look bad. I'm open to changing it to not having the 'Recipe' part.